ui/hooks/useAssetDetails.js

```diff
@@ -38,6 +39,21 @@ export function useAssetDetails(tokenAddress, userAddress, transactionData) {
         transactionData,
         nfts,
       );
+      const token = util.getContractAtAddress(tokenAddress);
```

Comment:
Yes we don't want to make these calls from the front end. We want to do these in the core/controllers

Reply:
It turns out this particular collection is an ERC-1155, so I wonder if the fix is for [`ERC1155Standard.getDetails`](https://github.com/MetaMask/core/blob/59aead98fb3b900c12fd47bdc9b49fd1dcae026d/packages/assets-controllers/src/Standards/NftStandards/ERC1155/ERC1155Standard.ts#L156) to query `symbol()` and `name()` like `ERC721Standard.getDetails` does.

Reply:
It turns out this particular contract is actually an ERC-1155 (not ERC-721).  The [ERC-1155 spec](https://eips.ethereum.org/EIPS/eip-1155) explicitly chose to not include `name()` or `symbol()`:

> The symbol function (found in the ERC-20 and ERC-721 standards) was not included as we do not believe this is a globally useful piece of data 

> The name function (for human-readable asset names, on-chain) was removed from the standard to allow the Metadata JSON to be the definitive asset name

However this particular ERC-1155 contract implemented them anyway.  But we should not rely on them generally.  Instead, I would look at improving [`ERC1155Standard.getDetails`](https://github.com/MetaMask/core/blob/59aead98fb3b900c12fd47bdc9b49fd1dcae026d/packages/assets-controllers/src/Standards/NftStandards/ERC1155/ERC1155Standard.ts#L190) to return the name from the `uri()` metadata.

Reply:
Hmmm `uri()` requires a token id, but we're setting approval for the whole collection.  So we don't have a token id to provide, and we never enter the [if block that calls `uri()`](https://github.com/MetaMask/core/blob/59aead98fb3b900c12fd47bdc9b49fd1dcae026d/packages/assets-controllers/src/Standards/NftStandards/ERC1155/ERC1155Standard.ts#L172).

It's not clear how best to get a collection name from ERC-1155.  Two ideas:

- Try to call `name()` / `symbol()` in case they're implemented (research whether they commonly are)
- Assume that a token id 0 exist and extract a collection name from `uri(0)`

Reply:
We do know conventions develop outside the actual specs, like how OpenSea's metadata format is more popular than the examples suggested in ERC-721 and ERC-1155.

One other source for collection level information is [`contractURI()`](https://docs.opensea.io/docs/contract-level-metadata), which is supported by OpenSea but not part of the standards.

OpenSea's [sample ERC-1155](https://github.com/ProjectOpenSea/opensea-erc1155/blob/c6f4573a1d9b830a560be32210d692222a1758e8/contracts/ERC1155Tradable.sol#L27-L30) also implements `name()` and `symbol()`.

We do have a dataset of [NFT contracts on mainnet](https://github.com/bergeron/nft-metadata/tree/main/contracts) if you want to filter for the ERC-1155s and see how many implement `name()`, `symbol()`, and `contractURI()`.

Reply:
hey brian,
thnx for your comment , i started doing it with your first suggestion ( Try to call name() / symbol() in ERC1155Standard.getDetails in case they're implemented ) , but like you said i don't know how common they are.
my idea was to call `name()` and `symbol()`, if we cannot found anything then we will display a generic name.

for this particular example i know that if i call `name()` and `symbol()` this will work but i'm not sure this will work for all the collections.

i'm open to any suggestion but i made this PR in WIP yesterday because i wanted to go deeper and do the fix in the core compoment

Reply:
hey @bergeron @adonesky1 
making changes to the core component means making changes to the @metamask-eth-abis package.
in fact, we use an abi that contains a set of functions ( ApprovalForAll , TransferBatch ... ) , but as the two functions `name()` and `symbol()` are not included in the [erc1155](https://eips.ethereum.org/EIPS/eip-1155) standard, this excludes the possibility of adding these two functions to the abi mentioned above.
so calling the two functions `name()` and `symbol()` from the core component will always return a rejected promise for a `function not found`.
all these elements lead me to conclude that the only possible change will be in the front end code.